core/blockchain_reader.go

```diff
@@ -343,3 +348,31 @@ func (bc *BlockChain) SubscribeAcceptedLogsEvent(ch chan<- []*types.Log) event.S
 func (bc *BlockChain) SubscribeAcceptedTransactionEvent(ch chan<- NewTxsEvent) event.Subscription {
 	return bc.scope.Track(bc.txAcceptedFeed.Subscribe(ch))
 }
+
+func (bc *BlockChain) GetFeeConfigAt(parent *types.Header) (commontype.FeeConfig, error) {
+	config := bc.Config()
+	bigTime := new(big.Int).SetUint64(parent.Time)
+	if !config.IsFeeConfigManager(bigTime) {
+		log.Debug("feeConfigManager is not activated, returning fees from config")
+		return config.FeeConfig, nil
+	}
+
+	if feeConfig, ok := bc.feeConfigCache.Get(parent.Root); ok {
+		return feeConfig.(commontype.FeeConfig), nil
+	}
+
+	stateDB, err := bc.StateAt(parent.Root)
+	if err != nil {
+		log.Debug("feeConfigManager is activated, but no state found; returning err", "err", err)
```

Comment:
Unless we require that the fee config is set when the precompile activates, this is probably a little excessive as well.

Reply:
What do you think of passing in the state instead of the parent header and then removing the error from the return value? I think this would make it easier to see exactly what is considered an error (retrieving the state) as opposed to retrieving the fee config from the state.

Reply:
Ya, this is better as is.